src/hooks/useTransactionSteps.ts

```diff
@@ -54,6 +59,17 @@ const useTransactionSteps = ({ transaction, onComplete }: Props) => {
     bestBlockFinalized: bestBlockFinalizedOnTarget
   } = useDashboard(targetRole);
 
+  useEffect(() => {
+    const getNonceOfTargetFinalizedBlock = async () => {
```

Comment:
4) Having now the `bestBlockFinalizedOnTarget` related with message delivered and the function to get the related nonce, i'm able to get it and save in the state ( to be moved this during the refactor to a global variable as indicated in previous PR)